docs/book/src/topics/secret-auto-rotation.md

```diff
@@ -0,0 +1,48 @@
+# Auto rotation of mounted contents and synced Kubernetes Secrets
+
+- **Design doc:** [Rotation Design](https://docs.google.com/document/d/1RGT0vmeUnN71n_u5fZKsSCa2YQpGw99rfGN9RlFMgHs/edit?usp=sharing)
+- **Feature State:** Secrets Store CSI Driver v0.0.15 [**alpha**] 
+
+When the secret/key is updated in external secrets store after the initial pod deployment, the updated secret will be periodically updated in the pod mount and the Kubernetes Secret.
+
+Depending on how the application consumes the secret data:
+1. **Mount Kubernetes secret as a volume:** Use auto rotation feature + Sync K8s secrets feature in Secrets Store CSI Driver, application will need to watch for changes from the mounted Kubernetes Secret volume. When the Kubernetes Secret is updated by the CSI Driver, the corresponding volume contents are automatically updated.
+2. **Application reads the data from container’s filesystem:** Use rotation feature in Secrets Store CSI Driver, application will need to watch for the file change from the volume mounted by the CSI driver.
+3. **Using Kubernetes secret for environment variable:** The pod needs to be restarted to get the latest secret as environment variable. 
+   1. Use something like https://github.com/stakater/Reloader to watch for changes on the synced Kubernetes secret and do rolling upgrades on pods
+
+## Enable auto rotation
+
+> NOTE: This alpha feature is not enabled by default.
+
+To enable auto rotation, enable the `--enable-secret-rotation` feature gate for the `secrets-store` container in the Secrets Store CSI Driver pods. The rotation poll interval can be configured using `--rotation-poll-interval`. The default rotation poll interval is `2m`. If using helm to install the driver, set `enableSecretRotation: true` and configure the rotation poll interval by setting `rotationPollInterval`. The rotation poll interval can be tuned based on how frequently the mounted contents for all pods and Kubernetes secrets need to be resynced to the latest.
+
+- The Secrets Store CSI Driver will update the pod mount and the Kubernetes Secret defined in `secretObjects` of SecretProviderClass periodically based on the rotation poll interval to the latest value.
+- If the `SecretProviderClass` is updated after the pod was initially created
+  - Adding/deleting objects and updating keys in existing `secretObjects` - the pod mount and Kubernetes secret will be updated with the new objects added to the `SecretProviderClass`.
+  - Adding new `secretObject` to the existing `secretObjects` - the Kubernetes secret will be created by the controller.
```

Comment:
Might be good to also callout this scenario and the expected behavior when sync is enabled without rotation.

Reply:
This is already in known limitations but modified that to call this out: https://deploy-preview-399--kubernetes-sigs-secrets-store-csi-driver.netlify.app/known-limitations.html